src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnRunner.java

```diff
@@ -124,6 +124,7 @@ public SpawnResult exec(Spawn spawn, SpawnExecutionPolicy policy)
     TreeNode inputRoot = repository.buildFromActionInputs(inputMap);
     repository.computeMerkleDigests(inputRoot);
     Command command = buildCommand(spawn.getArguments(), spawn.getEnvironment());
+    boolean uploadLocalResults = options.remoteUploadLocalResults;
```

Comment:
If this is indeed required, I would much rather add a whole new flag. Nobody ever would guess that noremote_upload_local_results would do that.
Also, since you are changing the action key in this case, the proper thing to do is to start a different context for the RemoteExecutionMetadata. As is, you're passing in the previous action key, which is not correct anymore.